src/mawpy/steps/incremental_clustering.py

```diff
@@ -53,33 +52,32 @@ def _get_cluster_center(row, mapping, dur_constr):
         return lat_long[0], lat_long[1], unc
 
 
-def _mean_ignore_minus_ones(series):
+def _mean_ignore_minus_ones(series: pd.Series) -> float:
     """
         Calculates the mean of the column without considering -1 entries in the column
     """
     return series[series != -1].mean()
 
 
-def _merge_stays(stay_to_update, updated_stay, df_by_user, group_avgs, group_avgs_index_to_update):
+def _merge_stays(stay_to_update: int, updated_stay: int, df_by_user: pd.DataFrame, group_avgs: pd.DataFrame,
+                 group_avgs_index_to_update: int) -> pd.DataFrame:
     """
         Merges two stays for a user and updates the mean_lat and mean_long of the stay.
     """
     df_by_user.loc[df_by_user[STAY] == stay_to_update, STAY] = updated_stay
-    merged_values = df_by_user[df_by_user[STAY] == updated_stay][[STAY_LAT, STAY_LONG]]
+    merged_values = df_by_user[df_by_user[STAY] == updated_stay][STAY_LAT_LONG]
     new_avg = merged_values.apply(_mean_ignore_minus_ones).fillna(-1)
 
-    group_avgs.loc[group_avgs[STAY] == updated_stay, [STAY_LAT, STAY_LONG]] = new_avg.values
-    df_by_user.loc[df_by_user[STAY] == updated_stay, STAY_LAT] = new_avg.values[0]
-    df_by_user.loc[df_by_user[STAY] == updated_stay, STAY_LONG] = new_avg.values[1]
+    group_avgs.loc[group_avgs[STAY] == updated_stay, STAY_LAT_LONG] = new_avg.values
+    df_by_user.loc[df_by_user[STAY] == updated_stay, STAY_LAT_LONG] = new_avg.values
 
     group_avgs.loc[group_avgs_index_to_update, STAY] = updated_stay
-    group_avgs.loc[group_avgs_index_to_update, STAY_LAT] = new_avg.values[0]
-    group_avgs.loc[group_avgs_index_to_update, STAY_LONG] = new_avg.values[1]
+    group_avgs.loc[group_avgs_index_to_update, STAY_LAT_LONG] = new_avg.values
 
     return df_by_user
 
 
-def _get_combined_stay(df_by_user, threshold=1):
+def _get_combined_stay(df_by_user: pd.DataFrame, threshold: float = 0.2) -> pd.DataFrame:
```

Comment:
Just checking if the change in the threshold value is intentional?

Reply:
I asked UW ThinkLab to confirm the value in the last meeting, I have added a TODO as well now.